database/memorydb.go

```diff
@@ -0,0 +1,81 @@
+package database
+
+import (
+	"fmt"
+	"sync"
+)
+
+type todos map[int]string
+
+type Database struct {
+	mu     sync.Mutex
+	data   todos
+	lastId int
+}
+
+func CreateDatabase() *Database {
+	return &Database{data: make(todos), lastId: -1}
+}
+
+func (db *Database) FetchTodo(id int) (string, error) {
+	var err error
+	db.mu.Lock()
```

Comment:
There's no need to lock the DB for reading

Reply:
Wouldn't that cause an Unexpected behavior or a race condition? 
From what I thought, if there's a call to remove and another one to retrieve the same ID at the same time, it won't be known which one will go first, so it can either return the user or not

Reply:
`sync.RWMutex` is designed for these kinds of situations. It allows multiple readers to hold the lock at the same time, but only a single writer.
If you were using `sync.Mutex` and a write operation happens at the same time, the read operation could see inconsistent or unexpected results, which is generally not what you want.

Consider this scenario:
- Goroutine A starts the FetchAll() function and acquires the read lock.
- Meanwhile, Goroutine B tries to delete an element from the map, which requires acquiring a write lock.
- Because Goroutine A holds a read lock, Goroutine B cannot acquire the write lock and has to wait.
- Goroutine A finishes its operation and releases the read lock.
- Goroutine B can now acquire the write lock and delete the element from the map.

Does this clarify your question?

Reply:
Thanks! 

I'm going to switch to `sync.RWMutex`, as what I understand using it will make me keep it on read operations, right?

Reply:
Yes, sorry, I thought that's what was already being used in the code.

Reply:
Fixed 4dd1c82007fc0b89bc231c7e5277aa30cedb2b43